adapters/adyoulike/adyoulike_test.go

```diff
@@ -0,0 +1,246 @@
+			},
+		},
+	}
+
+	expectedResponse := adapters.BidderResponse{
+		Currency: "",
+		Bids: []*adapters.TypedBid{
+			{
+				Bid: &openrtb.Bid{
+					ID:    "12340",
+					ImpID: "10",
+					Price: 300,
+					NURL:  "http://example.com/winnoticeurl0",
+					AdM:   "%3C%3Fxml%20version%3D%221.0%22%20encod%2Fhtml%3E",
+				},
+				BidType: "banner",
+			},
+			{
+				Bid: &openrtb.Bid{
+					ID:    "12341",
+					ImpID: "11",
+					Price: 301,
+					NURL:  "http://example.com/winnoticeurl1",
+					AdM:   "%3C%3Fxml%20version%3D%221.0%22%20encod%2FVAST%3E",
+				},
+				BidType: "video",
+			},
+			{
+				Bid: &openrtb.Bid{
+					ID:    "12342",
+					ImpID: "12",
+					Price: 302,
+					NURL:  "http://example.com/winnoticeurl2",
+					AdM:   "{'json':'response','for':'native'}",
+				},
+				BidType: "native",
+			},
+		},
+	}
+
+	bidder, buildErr := Builder(openrtb_ext.BidderAdyoulike, config.Adapter{
+		Endpoint: testsBidderEndpoint,
+	})
+
+	assert.Nil(t, buildErr, "buildErr must be nil")
+
+	httpResponse := &adapters.ResponseData{StatusCode: http.StatusOK, Body: responseBody}
+	bidResponse, errs := bidder.MakeBids(&request, nil, httpResponse)
+
+	if len(bidResponse.Bids) != 3 {
+		t.Fatalf("Expected 3 bids. Got %d", len(bidResponse.Bids))
+	}
+	if len(errs) != 0 {
+		t.Errorf("Expected 0 errors. Got %d", len(errs))
+	}
+
+	for i, typedBid := range bidResponse.Bids {
+
+		expected := expectedResponse.Bids[i].Bid
+
+		assert.Equal(t, expected.ID, typedBid.Bid.ID, "Incorrect Bid.id")
+		assert.Equal(t, expected.ImpID, typedBid.Bid.ImpID, "Incorrect Bid.impid")
+		assert.Equal(t, expected.Price, typedBid.Bid.Price, "Incorrect Bid.price")
+		assert.Equal(t, expected.NURL, typedBid.Bid.NURL, "Incorrect Bid.nurl")
+		assert.Equal(t, expected.AdM, typedBid.Bid.AdM, "Incorrect Bid.adm")
+
+		assert.Equal(t, expectedResponse.Bids[i].BidType, typedBid.BidType, "Incorrect BidType")
+	}
+
+}
+
+func TestOpenRTBSurpriseResponse(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderAdyoulike, config.Adapter{
+		Endpoint: testsBidderEndpoint})
+
+	assert.Nil(t, buildErr, "buildErr must be nil")
+
+	bidResponse, errs := bidder.MakeBids(nil, nil,
+		&adapters.ResponseData{StatusCode: http.StatusNoContent, Body: []byte("")})
+	if bidResponse != nil && errs != nil {
+		t.Fatalf("Expected no bids and no errors. Got %d bids and %d", len(bidResponse.Bids), len(errs))
+	}
+
+	bidResponse, errs = bidder.MakeBids(nil, nil,
+		&adapters.ResponseData{StatusCode: http.StatusServiceUnavailable, Body: []byte("")})
+	if bidResponse != nil || len(errs) != 1 {
+		t.Fatalf("Expected one error and no bids. Got %d bids and %d", len(bidResponse.Bids), len(errs))
+	}
+
+	bidResponse, errs = bidder.MakeBids(nil, nil,
+		&adapters.ResponseData{StatusCode: http.StatusOK, Body: []byte("{:'not-valid-json'}")})
+	if bidResponse != nil || len(errs) != 1 {
+		t.Fatalf("Expected one error and no bids. Got %d bids and %d", len(bidResponse.Bids), len(errs))
+	}
+}
```

Comment:
Could you please represent your test cases as json files? The json framework is a powerful framework for adapter tests which removes the need for large blocks of embedded json, as you're building in `TestOpenRTBStandardResponse`. Additionally, we're building new memory leak assertions to the json framework in the future as the race condition tests haven't proved reliable enough.

Reply:
done.